File: day04/solution.js
let crypto = require("crypto");

function getMD5HexHash(input) {
	return crypto.createHash('md5').update(input).digest('hex');
}

let secretKey = "yzbqklnj";

let i = 0;
let currentHash = "";

let zeros5 = "0".repeat(5);
let zeros6 = "0".repeat(6);

while (!currentHash.startsWith(zeros5)) {
	i++;
	currentHash = getMD5HexHash(secretKey + i);

	if (currentHash.startsWith(zeros6)) {
		console.log("Part 2:", i);
	}
}

console.log("Part 1:", i);

while (!currentHash.startsWith(zeros6)) {
	i++;
	currentHash = getMD5HexHash(secretKey + i);
}

console.log("Part 2:", i);

File: day04/solution.py
import hashlib

def getMD5(str):
	return hashlib.md5(str).hexdigest()

secretKey = "iwrupvqb"

fiveZeros = '0' * 5
sixZeros = '0' * 6

i = 0
currentHash = ""
while not currentHash.startswith(fiveZeros):
	i += 1
	currentHash = getMD5(secretKey + str(i))

print "Minimum value of n to start with 5 zeros: " + str(i) + " " + currentHash

i = 0
currentHash = ""
while not currentHash.startswith(sixZeros):
	i += 1
	currentHash = getMD5(secretKey + str(i))

print "Minimum value of n to start with 6 zeros: " + str(i)
